ada/language/ast.py

```diff
@@ -7586,7 +7586,7 @@ def check_for_type(typ=T.BaseTypeDecl.entity):
 
             Entity.parent.cast(T.CallExpr).then(
                 lambda ce: ce.check_for_type(
-                    bind_origin(Self, typ.expr_type)
+                    bind_origin(Self, typ.comp_type(True))
```

Comment:
plz use named notation for bool params. Also, all the rest of this property seems comment on at least which use case we are in, and this is not the case here for the default case. Could you also add a comment ? kthx